docker/app.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+from flask import Flask, request
+import logging
+
+logging.basicConfig(filename="recommendations.log", level=logging.INFO, filemode='w')
+logging.getLogger().addHandler(logging.StreamHandler())
+logging.info("App started.")
+
+import recommender_system
+import reader
+from constants import Method, METHODS_CNT
+
+app = Flask(__name__)
+
+
+@app.route('/tips/v1', methods=['POST'])
+def do_recommendation():
+    content = request.get_json()
+    logging.info("JSON request received. ")
+
+    try:
+        bucket, user_events, tips = reader.read_request_json(content)
+    except Exception:
+        logging.info(content)
+        return
+
+    logging.info("User events and tips read.")
+    
+    method_int = bucket % METHODS_CNT
+    method = Method(method_int)
+    logging.info("Current recommendation algorithm id: " + str(method.name))
+    
+    recommendation = recommender_system.recommend(user_events, tips, method)
+    logging.info("Recommendation made: " + str(recommendation))
+
+    data = {"showingOrder": recommendation,
```

Comment:
I just realized that we also should also send a version of the algorithm we use. Let's add one more field to this JSON: `version`

Reply:
We can use the following version format: `major.minor`. For example `0.1`

Reply:
(this is not urgent - it's possible to adapt code for the missing property and we don't need to re-deploy code asap).

Reply:
Done.